src/sparql/TripleBuilder.ts

```diff
@@ -32,19 +43,20 @@ export default class TripleBuilder {
 					value: 'anyValue' + condition.propertyId,
 				};
 			case ( PropertyValueRelation.Matching ):
-				return this.buildTripleForExplicitValue( condition.datatype, condition.value );
+				return this.buildTripleForExplicitValue( condition.datatype, this.checkValueType( condition.value ) );
 			default:
 				throw new Error( `unsupported relation: ${condition.propertyValueRelation}` );
 		}
 	}
 
-	private buildTripleForExplicitValue( datatype: string, value: string ): Term {
+	private buildTripleForExplicitValue( datatype: string, value: string | number ): Term {
 		switch ( datatype ) {
 			case 'string':
+			case 'quantity':
```

Comment:
This line is probably causing the browser test failure. We still want to treat `quantity` as a datatype with only limited support in this pull request. We'll build the SPARQL in https://phabricator.wikimedia.org/T276938